lib/rex/socket.rb

```diff
@@ -732,7 +732,15 @@ def getlocalname
   # Return peer connection information.
   #
   def getpeername
-    return Socket.from_sockaddr(super)
+    peer_name = nil
+    begin
+      peer_name = Socket.from_sockaddr(super)
+    rescue ::Exception => e
```

Comment:
Normally, rescuing all the exceptions isn't desired. If you're trying to rescue some specific exceptions, and you know what they are, I'd say better to be more specific.